### PR TITLE
Added New years gift home task for 04.11.2023  + Hometask VegetableSalad

### PR DESCRIPTION
**Pull request by Vitaliiitk**
Було достатньо складнощів, треба коменти. Дякую.

### CODE REVIEW

**Comment by Vitaliiitk** [comment]
Додав дз Vegetablesalad, не дуже зрозуміло, як користуватися IoC, прошу покоментати, як можна зробити, щоб реалізувати IoC у VegetableSalad hometask. Дякую.

File: VegetableSalad/VegetableSalad/Services/CreatePlant.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        public double Weight { get; set; }
+        public string Taste { get; set; }
+        public ushort CaloriesPerGram { get; set; }
+
+        public void ParameterOfPlan()
+        {
+            Name = Console.ReadLine();
+            Random rnd = new Random();
+
+            if (Name == "cucumber")
```
Comment (petrochernykh) [inline VegetableSalad/VegetableSalad/Services/CreatePlant.cs:19]: I think it's better to create class of a concrete vegetable what will inherit BaseVegetable

File: VegetableSalad/VegetableSalad/Services/FindVegetable.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿namespace VegetableSalad.Services
+{
+    internal class FindVegetable
+    {
+        public void FindTheMostCaloricVegetable(MakeSalad saladDish)
+        {
+            var mySalad = saladDish.MySalad();
+            double caloriesOfVegetable = (double)(mySalad[0].CaloriesPerGram) / (double)(100) * mySalad[0].Weight;
```
Comment (petrochernykh) [inline VegetableSalad/VegetableSalad/Services/FindVegetable.cs:8]: Can be moved to a separate method

File: VegetableSalad/VegetableSalad/Services/FindVegetable.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            int index = 0;
+
+            foreach (var item in mySalad)
+            {
+                double calories = (double)(item.CaloriesPerGram) / (double)(100) * item.Weight;
```
Comment (petrochernykh) [inline VegetableSalad/VegetableSalad/Services/FindVegetable.cs:13]: What's the difference between this  **double caloriesOfVegetable** and  **double calories**  variables

File: VegetableSalad/VegetableSalad/Services/SortVegetables.cs
Comment (petrochernykh) [inline VegetableSalad/VegetableSalad/Services/SortVegetables.cs:1]: I see that you have created  one class per one action. You should create one big SaladService and put all methods that perform any actions with salad in it

File: VegetableSalad/VegetableSalad/Services/SortVegetables.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿
+
+namespace VegetableSalad.Services
+{
+    internal class SortVegetables
+    {
+        public void SortVegetablesInSaladByWeight(MakeSalad saladDish)
+        {
+            var mySalad = saladDish.MySalad();
+            var enum1 = from vegetable in mySalad orderby vegetable.Weight select vegetable;
```
Comment (petrochernykh) [inline VegetableSalad/VegetableSalad/Services/SortVegetables.cs:10]: It's better to use extension mehtods. Vegetables.OrderBy().Select()

File: VegetableSalad/VegetableSalad/Services/SortVegetables.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            Console.WriteLine("From lightest  to heviest vegetable weight:");
+            foreach (var e in enum1)
+            {
+                Console.WriteLine(e.Name + " has weight " + e.Weight + " gram");
+            }
+            Console.WriteLine("From heviest to lightest vegetable wight:");
+            var enum2 = from vegetable in mySalad orderby vegetable.Weight descending select vegetable;
```
Comment (petrochernykh) [inline VegetableSalad/VegetableSalad/Services/SortVegetables.cs:17]: Actually, this is not Enum

**Pull request merged**